### ISSUE TITLE
[Bug] New-DbaDbTable: Column Length not set correctly for string data type.

### ISSUE THREAD

**Issue by bertvanlandeghem** [sqlcollaborative/dbatools#5882]
### Environmental information
#### PowerShell version:

Name                           Value                                                                                                                                                                                                                     
----                           -----                                                                                                                                                                                                                     
PSVersion                      5.1.15063.1805                                                                                                                                                                                                            
PSEdition                      Desktop                                                                                                                                                                                                                   
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                   
BuildVersion                   10.0.15063.1805                                                                                                                                                                                                           
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                           
WSManStackVersion              3.0                                                                                                                                                                                                                       
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                       
SerializationVersion           1.1.0.1                                                                                                                                                                                                                   

#### dbatools Module version:
Name    : dbatools
Path    : C:\Program Files\WindowsPowerShell\Modules\dbatools\1.0.19\dbatools.psd1
Version : 1.0.19

#### SQL Server: 
Microsoft SQL Server 2017 (RTM-GDR) (KB4494351) - 14.0.2014.14 (X64)   Apr  5 2019 09:18:51   Copyright (C) 2017 Microsoft Corporation  Developer Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) (Hypervisor) 

### Report
New-DbaDbTable:  Column Length not set correctly for string data type.


#### Host used
- [ ] powershell.exe
- [ ] ISE
- [x] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [x] Still buggy in powershell.exe

#### Errors Received
No errors received, column length for VarChar column is set to 1 in SQL Server, as if the length was not specified or 0.

#### Steps to Reproduce
$SqlInstance = 'MyInstance'
$DatabaseName = 'MyDatabase'
$TableName = 'Documents'

$Columns = @()
$Columns += @{ Name = 'testCol'; Type = 'VarChar'; MaxLength = 20; Nullable  = $false }

$MyTable = New-DbaDbTable -SqlInstance $SqlInstance -Database $DatabaseName -Name $TableName -ColumnMap $Columns
$MyTable.Columns['testCol'].DataType


Name                  : varchar
SqlDataType           : VarChar
Schema                :
MaximumLength         : 0               <--- should be 20
NumericPrecision      : 0
NumericScale          : 0
XmlDocumentConstraint : Default
IsNumericType         : False
IsStringType          : True


#### Expected Behavior
MaximumLength of string data type should be 20

#### Actual Behavior

MaximumLength of string data type is 0


IMHO the problem is situated in:

https://github.com/sqlcollaborative/dbatools/blob/c32c95d0ddd9db43bbc1091707934fb332ca2d2c/functions/New-DbaDbTable.ps1#L339

```
$dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType, $maxlength
```
which should be:
```
$dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType, $column.MaxLength
```
Isolating the foreach loop containing L339 loop shows the behavior:
```
$Columns = @()
$Columns += @{ Name = 'testCol'; Type = 'VarChar'; MaxLength = 20; Nullable  = $false }

foreach ($column in $Columns) {
    $sqlDbType = [Microsoft.SqlServer.Management.Smo.SqlDataType]$($column.Type)
    if ($sqlDbType -eq 'VarBinary' -or $sqlDbType -eq 'VarChar') {
        if ($column.MaxLength -gt 0) {

            $dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType, $maxlength
        } else {
            $sqlDbType = [Microsoft.SqlServer.Management.Smo.SqlDataType]"$(Get-SqlType $column.DataType.Name)Max"
            $dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType
        }
    } else {
        $dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType
    }
    $dataType
}


foreach ($column in $Columns) {
    $sqlDbType = [Microsoft.SqlServer.Management.Smo.SqlDataType]$($column.Type)
    if ($sqlDbType -eq 'VarBinary' -or $sqlDbType -eq 'VarChar') {
        if ($column.MaxLength -gt 0) {

            $dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType, $column.MaxLength
        } else {
            $sqlDbType = [Microsoft.SqlServer.Management.Smo.SqlDataType]"$(Get-SqlType $column.DataType.Name)Max"
            $dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType
        }
    } else {
        $dataType = New-Object Microsoft.SqlServer.Management.Smo.DataType $sqlDbType
    }
    $dataType
}


Name                  : varchar
SqlDataType           : VarChar
Schema                :
MaximumLength         : 0
NumericPrecision      : 0
NumericScale          : 0
XmlDocumentConstraint : Default
IsNumericType         : False
IsStringType          : True

Name                  : varchar
SqlDataType           : VarChar
Schema                :
MaximumLength         : 20
NumericPrecision      : 0
NumericScale          : 0
XmlDocumentConstraint : Default
IsNumericType         : False
IsStringType          : True

```

**Comment by potatoqualitee** [comment]
thanks so much, bert! appreciate the proposed solution as well. any interest in submitting the Pull request/fix?

**Comment by bertvanlandeghem** [comment]
> 
> 
> thanks so much, bert! appreciate the proposed solution as well. any interest in submitting the Pull request/fix?

OK, will do.

**Comment by bertvanlandeghem** [comment]
Closed by #5883.
